File: completed-labs/07/java/ListVisualiser.java
public class ListVisualiser {
	public SLList toVisualise;
	public String pointers = "";
	public String lines = "";
	public String values= "";

	// All instance of ListVisualiser share this atrs.
	public static String separatorLine = "->";
	public static String separatorSpace = "  ";
	public static String separatorTab = "    ";

    public ListVisualiser(SLList l) { 
		this.toVisualise = l;
    }
    public void visualise() {
        SLList.Node begin = this.toVisualise.l;
        while(begin.next.next!=null) {
			
			pointers += "[*|*]" + ListVisualiser.separatorLine;
			lines += "  |  " + ListVisualiser.separatorSpace;
			if((Integer) begin.info > 9 ) {
				values+= ListVisualiser.separatorSpace + begin.info + ListVisualiser.separatorTab.substring(0,ListVisualiser.separatorTab.length()-1);
			}else {
				values+= ListVisualiser.separatorSpace + begin.info + ListVisualiser.separatorTab;
			}
			
			
			begin = begin.next;
		}
		pointers += "[*|/]";
		lines += "  |  ";
        values+= "  " + begin.info;
        
        System.out.println(pointers);
		System.out.println(lines);
		System.out.println(values);

    }
}

File: completed-labs/07/cpp/ListVisualiser.cpp
#include "SLList.hpp"
#include "ListVisualiser.hpp"
#include <iostream>
using namespace std;
ListVisualiser::ListVisualiser(SLList *a) {

	mySLList = a;
}

void ListVisualiser::visualise() {  
	for (int i = 0; i < mySLList->length(); i++)
	{
		if (mySLList->nthRest(i)->rest() != SLList::NIL)
		{
			cout << "[*|*]->	";
		}
		else
		{
			cout << "[*|/]";
		}
	}
	cout << endl;

	
	for(int j = 0; j < mySLList->length(); j++)
	{
		cout << " |	";
	}
	
	cout << endl;

	for (int k = 0; k < mySLList->length(); k++)
	{
		cout << " " << mySLList->nth(k) << "	";
	}

	
	

}
